tests/terraform/checks/provider/panos/test_credentials.py

```diff
@@ -0,0 +1,20 @@
+import unittest
+
+from checkov.terraform.checks.provider.panos.credentials import check
+from checkov.common.models.enums import CheckResult
+
+
+class TestCredentials(unittest.TestCase):
+
+    def test_success(self):
+        provider_conf = {}
+        scan_result = check.scan_provider_conf(conf=provider_conf)
+        self.assertEqual(CheckResult.PASSED, scan_result)
+
+    def test_failure(self):
+        provider_conf = {'api_key' :['LUFRPT1yWFdyMFg5NlZxZ1ViU2ZhMTh6aGVEbDJ1UFU9ck9vc2tGcmlHV0tDbWRFa2cxcGUxSU8wMlVjaE9ReU0yYWN5SU1rL2pEOGhDcE50WEt5ABlHQWZoTm8xNG1SQQ==']}
```

Comment:
Please parse the resource config specified as a string, like here: https://github.com/bridgecrewio/checkov/blob/master/docs/6.Contribution/Contribute%20New%20Terraform%20Provider.md

The reason is that we have occasionally had issues where the results of parsing the string do not much the hardcoded resource conf, resulting in passing tests but failures in practice.

Although this is a very simple conf, and your definition is correct, it also protects against future parser changes.

Reply:
Thanks for reviewing, Mike. In the provider testing I created, I did copy the Linode `test_credentials.py` file, which is on the page of the link you provided, so they are almost identical, see below. Are you able to clarify what I need to change here? Many thanks!

Linode example:
```
    def test_success(self):
        provider_conf = {}

        scan_result = check.scan_provider_conf(conf=provider_conf)
        self.assertEqual(CheckResult.PASSED, scan_result)

    def test_failure(self):
        provider_conf = {'token' :['AKIAIOSFODNN7EXAMPLE']}
        scan_result = check.scan_provider_conf(conf=provider_conf)
        self.assertEqual(CheckResult.FAILED, scan_result)
```

PAN-OS:
```
    def test_success(self):
        provider_conf = {}
        scan_result = check.scan_provider_conf(conf=provider_conf)
        self.assertEqual(CheckResult.PASSED, scan_result)

    def test_failure(self):
        provider_conf = {'api_key' :['LUFRPT1yWFdyMFg5NlZxZ1ViU2ZhMTh6aGVEbDJ1UFU9ck9vc2tGcmlHV0tDbWRFa2cxcGUxSU8wMlVjaE9ReU0yYWN5SU1rL2pEOGhDcE50WEt5ABlHQWZoTm8xNG1SQQ==']}
        scan_result = check.scan_provider_conf(conf=provider_conf)
        self.assertEqual(CheckResult.FAILED, scan_result)
```

Reply:
@jamesholland-uk I created a new PR #2166 with updated docs on how to write the mentioned `hcl2` syntax. Other than that your changes look great and I'm happy to merge them after that change 🍾